tests: remove direct uses of *TestCluster

knz opened the pull request:
All commits except the last are from #107866 and prior.


Epic: CRDB-18499

knz committed:
tests: remove direct uses of `*TestCluster`

TestCluster has an interface. Use it.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/107986)
<!-- Reviewable:end -->

knz committed:
server: unexport Server

We should really only look at top level servers through an interface.

Release note: None
